dbt/include/global_project/macros/relations/postgres_relations.sql

```diff
@@ -0,0 +1,64 @@
+{% macro postgres__get_relations () -%}
+  {%- call statement('relations', fetch_result=True) -%}
+    -- {#
+    -- in pg_depend, objid is the dependent, refobjid is the referenced object
+    -- "a pg_depend entry indicates that the referenced object cannot be dropped without also dropping the dependent object."
+    -- #}
+    with relation as (
+        select
+            pg_rewrite.ev_class as class,
+            pg_rewrite.oid as id
+        from pg_rewrite
+    ),
+    class as (
+        select
+            oid as id,
+            relname as name,
+            relnamespace as schema,
+            relkind as kind
+        from pg_class
+    ),
+    dependency as (
+        select
+            pg_depend.objid as id,
+            pg_depend.refobjid as ref
+        from pg_depend
+    ),
+    schema as (
+        select
+            pg_namespace.oid as id,
+            pg_namespace.nspname as name
+        from pg_namespace
+        where nspname != 'information_schema' and nspname not like 'pg_%'
+    ),
+    relationships as (
+        select
+            referenced_class.name as referenced_name,
+            referenced_class.schema as referenced_schema_id,
+            dependent_class.name as dependent_name,
+            dependent_class.schema as dependent_schema_id,
+            referenced_class.kind as kind
+        from relation
+        join class as referenced_class on relation.class=referenced_class.id
+        join dependency on relation.id=dependency.id
+        join class as dependent_class on dependency.ref=dependent_class.id
+        where
+            referenced_class.kind in ('r', 'v') and
+            (referenced_class.name != dependent_class.name or
+             referenced_class.schema != dependent_class.schema)
+    )
+
+    select
```

Comment:
i'm not sure if this is significant, but is there any chance that these relationships can be duplicated? Like if you join a table to itself in a view:
```
-- models/downstream_view.sql

select t1.name, t2.name
from some_table t1
join some_table t2 on t1.parent_id = t2.child_id
```

Does that create two different entries in the internal relationships table between `downstream_view` and `some_table`? Regardless, might be worth distinct-ing the results here?

Reply:
nvm, you're grouping by all four so that will distinct the records